api/src/main/java/org/openmrs/module/fhir2/api/translators/impl/MedicationTranslatorImpl.java

```diff
@@ -0,0 +1,93 @@
+/*
+ * This Source Code Form is subject to the terms of the Mozilla Public License,
+ * v. 2.0. If a copy of the MPL was not distributed with this file, You can
+ * obtain one at http://mozilla.org/MPL/2.0/. OpenMRS is also distributed under
+ * the terms of the Healthcare Disclaimer located at http://openmrs.org/license.
+ *
+ * Copyright (C) OpenMRS Inc. OpenMRS is a registered trademark and the OpenMRS
+ * graphic logo is a trademark of OpenMRS Inc.
+ */
+package org.openmrs.module.fhir2.api.translators.impl;
+
+import javax.inject.Inject;
+
+import java.util.ArrayList;
+import java.util.Collection;
+
+import lombok.AccessLevel;
+import lombok.Setter;
+import org.hl7.fhir.r4.model.CodeableConcept;
+import org.hl7.fhir.r4.model.Coding;
+import org.hl7.fhir.r4.model.Medication;
+import org.openmrs.Drug;
+import org.openmrs.DrugIngredient;
+import org.openmrs.module.fhir2.FhirConstants;
+import org.openmrs.module.fhir2.api.FhirConceptService;
+import org.openmrs.module.fhir2.api.translators.MedicationTranslator;
+import org.springframework.stereotype.Component;
+
+@Component
+@Setter(AccessLevel.PACKAGE)
+public class MedicationTranslatorImpl implements MedicationTranslator {
+	
+	@Inject
+	private FhirConceptService conceptService;
+	
+	@Override
+	public Medication toFhirResource(Drug drug) {
+		if (drug == null) {
+			return null;
+		}
+		
+		Medication medication = new Medication();
+		medication.setId(drug.getUuid());
+		medication.setCode(new CodeableConcept().addCoding(new Coding(FhirConstants.MEDICATION_CODES_VALUE_SET_URI,
+		        drug.getConcept().getUuid(), drug.getConcept().getDisplayString())));
+		medication.setForm(new CodeableConcept().addCoding(new Coding(FhirConstants.MEDICATION_FORM_VALUE_SET_URI,
+		        drug.getDosageForm().getUuid(), drug.getDosageForm().getDisplayString())));
+		
+		Medication.MedicationIngredientComponent ingredient = new Medication.MedicationIngredientComponent();
+		CodeableConcept ingredientCode = new CodeableConcept();
+		for (DrugIngredient val : drug.getIngredients()) {
+			ingredientCode.addCoding(new Coding(FhirConstants.MEDICATION_INGREDIENT_VALUE_SET_URI,
+			        val.getIngredient().getUuid(), val.getIngredient().getDisplayString()));
+			
+			ingredient.setItem(ingredientCode);
+		}
+		medication.addIngredient(ingredient);
+		
+		return medication;
+	}
+	
+	@Override
+	public Drug toOpenmrsType(Drug existingDrug, Medication med) {
+		if (med == null) {
+			return null;
+		}
+		existingDrug.setUuid(med.getId());
+		
+		if (med.hasCode()) {
+			existingDrug
+			        .setConcept(conceptService.getConceptByUuid(med.getCode().getCoding().get(0).getCode()).orElse(null));
```

Comment:
Instead of this, use **conceptTranslator**
```java
@Inject
private ConceptTranslator conceptTranslator;

existingDrug.setConcept(conceptTranslator.toOpenmrsType(med.getCode()));
```